dev/AppNotifications/AppNotificationBuilder/AppNotificationProgressBar.cpp

```diff
@@ -0,0 +1,106 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+#include "pch.h"
+#include "AppNotificationProgressBar.h"
+#include "Microsoft.Windows.AppNotifications.Builder.AppNotificationProgressBar.g.cpp"
+
+namespace winrt::Microsoft::Windows::AppNotifications::Builder::implementation
+{
+    AppNotificationProgressBar::AppNotificationProgressBar()
+        :m_titleStatus{ BindMode::NotSet },
+         m_statusStatus{ BindMode::NotSet },
+         m_valueStatus{ BindMode::NotSet },
+         m_valueStringOverrideStatus{ BindMode::NotSet }
+    {};
+
+    void AppNotificationProgressBar::Title(winrt::hstring const& value)
+    {
+        m_title = value;
+        m_titleStatus = BindMode::Value;
+    }
+
+    void AppNotificationProgressBar::Status(winrt::hstring const& value)
+    {
+        m_status = value;
+        m_statusStatus = BindMode::Value;
+    }
+
+    void AppNotificationProgressBar::Value(double value)
+    {
+        THROW_HR_IF_MSG(E_INVALIDARG, value < 0.0 || value > 1.0, "You must provide a value between 0.0 and 1.0");
+
+        m_value = value; m_valueStatus = BindMode::Value;
+    }
+
+    void AppNotificationProgressBar::ValueStringOverride(winrt::hstring const& value)
+    {
+        m_valueStringOverride = value;
+        m_valueStringOverrideStatus = BindMode::Value;
+    }
+
+    winrt::Microsoft::Windows::AppNotifications::Builder::AppNotificationProgressBar AppNotificationProgressBar::SetTitle(winrt::hstring const& value)
+    {
+        Title(value);
+
+        return *this;
+    }
+
+    winrt::Microsoft::Windows::AppNotifications::Builder::AppNotificationProgressBar AppNotificationProgressBar::BindTitle()
+    {
+        m_titleStatus = BindMode::Bind;
```

Comment:
Can we append *BindMode to all this m_ variables?
For example m_tileBindMode, m_statusBindMode

Reply:
Sure

Reply:
Done